packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -179,6 +179,7 @@ type HTMLElementTheGoodParts = { toString: () => string } & Pick<
     | 'removeEventListener'
     | 'setAttribute'
     | 'setAttributeNS'
+    | 'shadowRoot'
```

Comment:
The `LightningElement` class in `base-lightning-element` refers to the view of the class from the inside (e.g. `this.refs`, `this.tagName`, etc.) – as opposed to `base-bridge-element` which refers to the view of the class from the outside (`elm.shadowRoot`, `elm.tagName`, etc.).

`this.shadowRoot` is actually not exposed in `LightningElement`s today. We have an open item to fix that, but it's an accurate representation of reality: https://github.com/salesforce/lwc/issues/3344

Reply:
Ah weird, I just realized that the property does exist, but it's always null:

https://github.com/salesforce/lwc/blob/372a37f6de47ac7fa163ab704359c922aeb6755f/packages/%40lwc/engine-core/src/framework/base-lightning-element.ts#L705-L709

This is a bit weird, but I guess it's fine, since in the standard `HTMLElement` the `shadowRoot` may be null as well.